### ISSUE TITLE
cannot compile demos?

### ISSUE THREAD

**Issue by nilsbecker** [backtracking/ocamlgraph#47]
hi, i just installed via opam and i'm unable to build the demos. i cwd into the build/ocamlgraph.1.8.6 directory in my opam tree, and then try `make demo`. this fails with 

```
File "examples/demo.ml", line 261, characters 38-39:  
Error: Signature mismatch:
       ...
       The type `edge' is required but not provided
```

`make demo.opt` as indicated in the README finds no build target.

**Comment by backtracking** [comment]
Hi,

> hi, i just installed via opam and i'm unable to build the demos. i cwd
> into the build/ocamlgraph.1.8.6 directory in my opam tree, and then try
> |make demo|. this fails with
> 
> |File "examples/demo.ml", line 261, characters 38-39:  
> Error: Signature mismatch:
>        ...
>        The type `edge' is required but not provided

I never tried to do that (i.e. to compile examples in opam's build
directory) but I guess it should work. So thanks for reporting this bug.

I confirm examples do compile without problem in the trunk.

> |make demo.opt| as indicated in the README finds no build target.

It should be 'make bin/demo.opt' instead. I just updated the README in
the trunk.

Sorry for the mess,
--
Jean-Christophe

**Comment by tiuno** [comment]
Hi,
    I've tried to compile demo with your indications `make bin/demo.opt` this doesn't work. It gives the exact same error message.
`$ make bin/demo.opt 
ocamlopt.opt -c -I src -I lib examples/demo.ml
File "examples/demo.ml", line 261, characters 38-39:
Error: Signature mismatch:
       ...
       The type `edge' is required but not provided
Makefile:559: recipe for target 'examples/demo.cmx' failed
make: *** [examples/demo.cmx] Error 2
`
However example/color.ml works fine.

**Comment by hugomg** [comment]
One thing I just noticed is that if I clone the repo and build the examples with `make examples` they will compile just fine but if I try to build them by hand with ocamlbuild then I get errors. If I go to the examples folder and run

    ocamlbuild -use-ocamlfind -pkg ocamlgraph -pkg graphics demo.native

I get an error complaining about the lack of a `Rand` module

    Error: Unbound module Rand

I believe this has to do with how the make version of the build uses `-I src`. Is there a way to avoid this? It would be nice to be able to build the examples using the same building commands system I will use for my real program.

Other than that, it would be nice if we could upload a more current version of ocamlgraph to opam. Right now the latest release (1.8.6) is over a year old and includes some interface changes that make it incompatible with the examples from the latest version (which is what people like me would find when they search the web for documentation).

**Comment by backtracking** [comment]
(sorry for the late answer)

> One thing I just noticed is that if I clone the repo and build the
> examples with |make examples| they will compile just fine but if I try
> to build them by hand with ocamlbuild then I get errors. If I go to the
> examples folder and run
> 
> |ocamlbuild -use-ocamlfind -pkg ocamlgraph -pkg graphics demo.native |
> 
> I get an error complaining about the lack of a |Rand| module
> 
> |Error: Unbound module Rand |

There was an extra "open Rand" in demo.ml.
I fixed that in commit 7f1f800d73.

-- 
Jean-Christophe

**Comment by backtracking** [comment]
> Other than that, it would be nice if we could upload a more current
> version of ocamlgraph to opam. Right now the latest release (1.8.6) is
> over a year old

You're right. I just made a new release 1.8.7 (as well as a pull request
for an opam package).

-- 
Jean-Christophe
